apis/v1alpha2/grpcroute_types.go

```diff
@@ -147,7 +147,7 @@ type GRPCRouteSpec struct {
 	//
 	// +optional
 	// +kubebuilder:validation:MaxItems=16
-	// +kubebuilder:default={{matches: {{method: {type: "Exact"}}}}}
+	// TODO(gauravkghildiyal): Discuss
```

Comment:
There seems to be some conflict between what we are validating (_"One or both of 'service' or 'method' must be specified"_) and what we are setting as the default (_"neither of the two"_)

I'm not sure if we can set a sensible default here. Does it mean we remove the defaulting? Or do we skip the validation?

Reply:
CC: @gnossen

Reply:
This was discussed in https://github.com/kubernetes-sigs/gateway-api/pull/2311#discussion_r1296286374

Reply:
Chatted with @gnossen and @gauravkghildiyal on this and came to the conclusion that removing this validation is correct. The behavior we want is "if someone specifies a match, the default type of that match should be exact." That's covered by this line: https://github.com/kubernetes-sigs/gateway-api/blob/ce5bf58ac9bf9101a09a9f64a9959a0f84802d8e/apis/v1alpha2/grpcroute_types.go#L325

The line we want to remove is this one: https://github.com/kubernetes-sigs/gateway-api/blob/ce5bf58ac9bf9101a09a9f64a9959a0f84802d8e/apis/v1alpha2/grpcroute_types.go#L150 That creates a default match of type "Exact" with no other config and is therefore failing validation and also not what we actually want here. Given that this default was never actually valid and this API is still experimental, I think it's good to remove the validation here - we should just add a release note to the PR description to document the change.